docs: update readme and create documentation PE-2332

luqven opened the pull request:
TSIA

- incorporates https://github.com/imgix/web-components/pull/49


<!---GHSTACKOPEN-->
### Stacked PR Chain: PE-2332
| PR | Title | Status |  Merges Into  |
|:--:|:------|:-------|:-------------:|
|#1|👉 docs: update readme and create documentation PE-2332|![](https://img.shields.io/github/pulls/detail/state/imgix/ix-video/1?label=Approved)|-|
|#2|docs: create ix-video hosted docs PE-2332|![](https://img.shields.io/github/pulls/detail/state/imgix/ix-video/2?label=Approved)|#1|
|#3|docs: rename vuepress to docs PE-2332|![](https://img.shields.io/github/pulls/detail/state/imgix/ix-video/3?label=Approved)|#2|
|#4|docs(release): v1.0.0-rc.1 PE-2332|![](https://img.shields.io/github/pulls/detail/state/imgix/ix-video/4?label=Approved)|#3|
<!---GHSTACKCLOSE-->

luqven committed:
docs: update readme and create documentation

luqven committed:
docs: create ix-video hosted docs

sherwinski commented on documentation/vue.md:
```diff
@@ -0,0 +1,71 @@
+
+However, Vue might show some warnings in the console if you don't declare your custom element in it's configuration. To turn these warnings off, you can let vue that the elements that start with `ix` are custom elements.
+
+If you're using Vite, you can add the following to your `vite.config.js`:
```
```suggestion
```

sherwinski commented on documentation/vue.md:
````diff
@@ -0,0 +1,71 @@
+  }
+```
+
+Or with Vite:
````
```suggestion
Or if you're using Vite, you can add the following to your `vite.config.js`:
```

sherwinski commented on documentation/vue.md:
````diff
@@ -0,0 +1,71 @@
+</template>
+```
+
+However, Vue might show some warnings in the console if you don't declare your custom element in it's configuration. To turn these warnings off, you can let vue that the elements that start with `ix` are custom elements.
````
```suggestion
You will need to signal to the Vue compiler that this component is a custom element by adding the following:
```

sherwinski commented on README.md:
```diff
@@ -1,215 +1,62 @@
+## Using `<ix-video>` on an HTML static page
 
-To lint the project run:
+Below is an example of how to import and use the `ix-video` custom element from a CDN on a static HTML page.
```
```suggestion
Below is an example of how to import and use the `ix-video` custom element from a direct import or CDN on a static HTML page.
```

sherwinski commented on README.md:
```diff
@@ -1,215 +1,62 @@
+Look at our docs to see how to us this inside your [React](/documentation/react.md), [Vue](/documentation/vue.md), or other [web-apps](/documentation/static.md).
 
-The rules are mostly the recommended rules from each project, but some have been turned off to make LitElement usage easier. The recommended rules are pretty strict, so you may want to relax them by editing `.eslintrc.json` and `tsconfig.json`.
+## Using `<ix-video>` on an HTML static page
```
Super nit, feel free to ignore but I thought this sounded a bit clearer (and consistent with L41)
```suggestion
## Using `<ix-video>` on a static HTML page
```

sherwinski commented on documentation/contributing.md:
```diff
@@ -0,0 +1,91 @@
+# Development
```
Personal preference but I think `CONTRIBUTING.md` should generally live in the project root

luqven replied:
💯 that mstm

sherwinski commented on documentation/attributes.md:
```diff
@@ -0,0 +1,34 @@
+### Attributes
```
Any reason to break attributes into its own nested doc? To me it feels like something that should be on the main README for immediate reference

luqven replied:
No strong opinion here. Happy to move it to the main README.

sherwinski commented on documentation/vue.md:
```diff
@@ -0,0 +1,71 @@
+          // treat any tag that starts with ix- as custom elements
+          isCustomElement: (name) => name.startsWith('ix-'),
+        }
+      }))
```
This was going to keep me up at night if I didn't say something
```suggestion
      })
    )
```

luqven committed:
docs: update vue docs

luqven committed:
chore: move contributing docs to root

luqven committed:
docs: move attribute docs to main readme

luqven committed:
docs: address feedback

luqven committed:
docs(vuepress): move usage section into ix-video md

The `usage.md` wasn't acutally linked/rendered by anything. Now the
section is rendered inside the ix-video document.

luqven committed:
chore: change wc references to ix-video

luqven committed:
chore: update installation for static usage

luqven committed:
chore: update hero image

luqven committed:
build: add docs dev script

luqven committed:
docs: move ix-video vue docs into sep md files

luqven committed:
docs: add vue installation docs reqs section

luqven committed:
docs: update vue readme to reflects new routes

luqven committed:
docs: update container example to highlight fixed

luqven committed:
docs: update vue routes

luqven committed:
docs: change cust h to h1 from h2

luqven committed:
docs: move md files into vuepress docs

luqven committed:
docs(readme): change docs links to vuepress directory

luqven committed:
docs: rename vuepress to docs

luqven committed:
docs: move bundle import to fix ssr issues

luqven committed:
build: add docs deploy script

luqven committed:
docs: fix broken quick-start links

luqven committed:
build: change docs scripts dir ref

luqven committed:
docs(release): v1.0.0-rc.1

luqven committed:
docs: update readme docs links

sherwinski commented on documentation/vue.md:
```diff
@@ -0,0 +1,70 @@
+# Using `≤ix-video>` custom element in a Vue app
```
Ah just noticed this while reviewing a different PR down the stack
```suggestion
# Using `<ix-video>` custom element in a Vue app
```

luqven committed:
docs: add warning and update build scripts
